app/scripts/metamask-controller.js

```diff
@@ -1561,7 +1563,7 @@ export default class MetamaskController extends EventEmitter {
         },
       },
       env: {
-        isAccountSyncingEnabled: isManifestV3,
+        isAccountSyncingEnabled: !isProduction() && isManifestV3,,
```

Comment:
Hmm. This `isProduction` function does not effectively check that the current build is a production build.

It only checks that this is _not_ "development" or "testing": https://github.com/MetaMask/metamask-extension/blob/9716e949259d7c89cb99f688aef670b8d7af4e73/shared/modules/environment.ts#L5

But there are many other environments: https://github.com/MetaMask/metamask-extension/blob/9716e949259d7c89cb99f688aef670b8d7af4e73/development/build/constants.js#L18

Reply:
This will be effective in disabling the check in production. But it will also disable it in any non-test builds produced on CircleCI, including release candidate builds, which would be confusing during release QA.

Reply:
Thoughts on setting this to `false` instead?

Reply:
> But it will also disable it in any non-test builds produced on CircleCI, including release candidate builds, which would be confusing during release QA.

Isn't that what we want, so that when v12.5.0 is tested, the feature is disabled?

Reply:
Fair enough, yes we would want those to be as production-like as possible. But our QA team uses a lot of different build types interchangeably, including development builds, so it would still be confusing.

What was the reason for leaving it enabled for development and e2e testing?

Reply:
>What was the reason for leaving it enabled for development and e2e testing?

I was just thinking not to inhibit the ongoing QA work. It can be more efficient to work with development builds while writing tests or doing exploratory testing.